src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -275,35 +276,43 @@
 
 (react/defc LoggedOut
   {:render
-   (fn [{:keys [props]}]
-     [:div {}
+   (fn [{:keys [this props]}]
+     ;; Google's code complains if the sign-in button goes missing, so we hide this component rather
+     ;; than removing it from the page.
+     [:div {:style {:display (when (:hidden? props) "none")}}
       [:div {:style {:marginBottom "2em"}} (text-logo)]
-      [:div {}
-       [sign-in/Button (select-keys props [:on-login])]]
+      [comps/Button {:text "Sign In" :onClick #(react/call :.handle-sign-in-click this)}]
       [:div {:style {:marginTop "2em" :maxWidth 600}}
        [:div {} [:b {} "New user? FireCloud requires a Google account."]]
        [:div {} "Please use the \"Sign In\" button above to sign-in with your Google Account.
          Once you have successfully signed-in with Google, you will be taken to the FireCloud
          registration page."]]
       [:div {:style {:maxWidth 600 :paddingTop "2em" :fontSize "small"}}
-        [Policy]]])})
-
-
-(defn- parse-hash-into-map []
-  ;; Transforms the browser navigation hash into a map to make it easier to pass around. Technique
-  ;; stolen from Swagger's authentication flow.
-  (let [hash (nav/get-hash-value)
-        hash (clojure.string/replace hash #"^[?]" "")
-        parts (clojure.string/split hash #"[&=]")
-        parts (map js/decodeURIComponent parts)]
-    (apply assoc {} parts)))
-
-
-(defn- attempt-auth [token on-done]
-  ;; Use basic ajax call here to bypass default ajax-orch failure behavior.
-  (utils/ajax {:url (str (config/api-url-root) "/me")
-               :headers {"Authorization" (str "Bearer " token)}
-               :on-done on-done}))
+       [Policy]]])
+   :component-did-mount
+   (fn [{:keys [props locals]}]
+     (swap! locals assoc :refresh-token-saved? true)
+     (let [{:keys [auth2 on-change]} props]
+       (-> auth2 (.-currentUser)
+           (.listen (fn [u]
+                      (on-change (.isSignedIn u) (:refresh-token-saved? @locals)))))))
```

Comment:
that's a huge - and positive - change! Needs to get documented somewhere, at minimum so QA understands.